### DISCUSSION TITLE
How to change tab space in clangd

### DISCUSSION THREAD

**Discussion by j0ker70** [clangd/clangd#1330] [General]
I have recently installed clangd-10 in my machine. I am trying neovim with builtin lsp to work with clangd. One problem I am facing is whenever I am trying to format my cpp code with `:lua vim.lsp.buf.format()`, it formats the file with 2 tab space. I my neovim settings, I have specified tab space to be 4. Somehow it is overriding with clangd. How can I change it?

**Comment by i-ky** [comment]
You need to configure desired formatting style using `.clang-format` file or by passing `--fallback-style` option to clangd. See [clang-format style options](https://clang.llvm.org/docs/ClangFormatStyleOptions.html), I guess you need `IndentWidth`.

**Reply by j0ker70** [reply]
How to set .clang-format globally? I mean I don't want to write for every project. I want this to be default.

**Reply by i-ky** [reply]
> **clang-format** for each input file will try to find the `.clang-format` file located in the closest parent directory of the input file

So you can put `.clang-format` into common parent directory of all your projects:
```
/path/to/projects/
|- .clang-format
|- project1/
|- project2/
|- ...
`- projectN/

**Reply by j0ker70** [reply]
Can I do something like in my home folder? Like this file would be in `$HOME/.clang-format`

**Reply by i-ky** [reply]
As far as I'm aware, no.

**Reply by j0ker70** [reply]
It worked. `$HOME/.clang-format` worked. Thank you

**Reply by HighCommander4** [reply]
The search for `.clang-format` files does go up the directory tree, so as long as your projects are somewhere in the `$HOME` directory tree, then `$HOME/.clang-format` should be read.

**Reply by j0ker70** [reply]
By the way, what is the default clang-format configuration? Is there any way to know that?

**Reply by i-ky** [reply]
I think it falls back to [LLVM coding style](https://llvm.org/docs/CodingStandards.html). If you use [`clang-format`](https://clang.llvm.org/docs/ClangFormat.html) directly, you can use `--dump-config` option to see the configuration it uses. I don't think you can get the same from `clangd`.
